### PR TITLE
Add arguments to MariadbOperator class

### PR DESCRIPTION
**Pull request by adolfo-ab**
##### Short description:
Add missing arguments to MariadbOperator

##### More details:
MariadbOperator is created through Helm, so class_generator wasn't able to pick up its arguments.

## Summary by CodeRabbit

- **New Features**
	- Enhanced configuration options for the MariaDB operator, allowing users to specify various parameters for improved customization.
	- Introduced a method to convert the operator's configuration into a dictionary format for easier management and integration.

These changes provide users with greater flexibility and control over the MariaDB operator's behavior and settings.

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: ocp_resources/mariadb_operator.py [hunk 1]
```diff
@@ -13,6 +13,128 @@ class MariadbOperator(NamespacedResource):
...
+            tolerations (List[Dict[str, Any]]): Tolerations for node taints.
+            webhook (Dict[str, Any]): Webhook configuration.
+        """
+        super().to_dict()
+
+        if not self.kind_dict and not self.yaml_file:
+            self.res["spec"] = {}
+            _spec = self.res["spec"]
+
+            if self.affinity is not None:
```
Comment (myakove) [inline ocp_resources/mariadb_operator.py:97]: none of them are mandatory?

File: ocp_resources/mariadb_operator.py [hunk 2]
```diff
@@ -13,8 +13,130 @@ class MariadbOperator(NamespacedResource):
...
+            if self.resources is not None:
+                _spec["resources"] = self.resources
+            if self.security_context is not None:
+                _spec["securityContext"] = self.security_context
+            if self.service_account is not None:
+                _spec["serviceAccount"] = self.service_account
+            if self.tolerations is not None:
+                _spec["tolerations"] = self.tolerations
+            if self.webhook is not None:
+                _spec["webhook"] = self.webhook
```
[the remark this answers is not in the document]
Comment (myakove) [reply ocp_resources/mariadb_operator.py:140]: The current code is more readable, no need to change it.

**Review by myakove — APPROVED** [review]

**Comment by adolfo-ab** [comment]
verified

**Pull request merged**